src/io/PipeWriter.zig

```diff
@@ -1003,7 +1003,12 @@ pub fn WindowsBufferedWriter(
             this.is_done = true;
             if (this.pending_payload_size == 0) {
                 // will auto close when pending stuff get written
-                this.close();
+                if (this.source) |source| {
+                    switch (source) {
+                        .sync_file, .file => {},
```

Comment:
this is confusing to me

Reply:
can you add a comment explainign why it should only call close for pipes and ttys?